rkt/images.go

```diff
@@ -224,49 +222,101 @@ func (f *fetcher) fetchSingleImage(img string, asc string, discover bool) (strin
 		return "", fmt.Errorf("unable to access %q: %v", img, err)
 	}
 
-	if discover && u.Scheme == "" {
+	switch u.Scheme {
+	// image name
+	case "":
+		return f.fetchImageByName(img, ascFile)
+	case "http", "https", "file", "docker":
+		return f.fetchImageByURL(u, ascFile)
+	}
+
+	return "", fmt.Errorf("rkt only supports image name and http, https, docker or file URLs (%s)", img)
```

Comment:
This particular function doesn't support image key (hash) as it's processed previously by `finder`. 
`finder` is called by `run` and `prepare` as they also accept image key while `fetch` directly calls `fetcher`.

Perhaps:
* `image name` or what will be its definition should be defined (appc/spec#479) and documented.
* Implement something like getImageType to return, just during argument parsing:
 * for `rkt fetch` `rkt fetch only supports image name and http, https, docker or file URLs`
 * for `rkt run/prepare` `rkt run/prepare only supports image name, image key and http, https, docker or file URLs`